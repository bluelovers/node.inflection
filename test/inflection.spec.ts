import inflection from '../index';

describe('test .indexOf', function ()
{
	it('should return proper index key', function ()
	{
		let actual: number;
		actual = inflection.indexOf(['hi', 'there'], 'guys');
		expect(actual).toStrictEqual(-1);
		expect(actual).toMatchSnapshot();

		actual = inflection.indexOf(['hi', 'there'], 'hi');
		expect(actual).toStrictEqual(0);
		expect(actual).toMatchSnapshot();

	});
});

describe('test .pluralize', function ()
{
	it('should pluralize the given word', function ()
	{
		let actual: string;
		actual = inflection.pluralize('people');
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('men');
		expect(actual).toStrictEqual('men');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('women');
		expect(actual).toStrictEqual('women');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('woman');
		expect(actual).toStrictEqual('women');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('person');
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('octopus');
		expect(actual).toStrictEqual('octopi');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('human');
		expect(actual).toStrictEqual('humans');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('aircraft');
		expect(actual).toStrictEqual('aircraft');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('luck');
		expect(actual).toStrictEqual('luck');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('Hat');
		expect(actual).toStrictEqual('Hats');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('life');
		expect(actual).toStrictEqual('lives');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('bath');
		expect(actual).toStrictEqual('baths');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('calf');
		expect(actual).toStrictEqual('calves');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('foot');
		expect(actual).toStrictEqual('feet');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('book');
		expect(actual).toStrictEqual('books');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('goose');
		expect(actual).toStrictEqual('geese');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('tooth');
		expect(actual).toStrictEqual('teeth');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('teeth');
		expect(actual).toStrictEqual('teeth');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('knife');
		expect(actual).toStrictEqual('knives');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('half');
		expect(actual).toStrictEqual('halves');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('cave');
		expect(actual).toStrictEqual('caves');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('save');
		expect(actual).toStrictEqual('saves');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('street');
		expect(actual).toStrictEqual('streets');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('streets');
		expect(actual).toStrictEqual('streets');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('data');
		expect(actual).toStrictEqual('data');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('meta');
		expect(actual).toStrictEqual('meta');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('summons');
		expect(actual).toStrictEqual('summonses');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('whereas');
		expect(actual).toStrictEqual('whereases');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('person', 'guys');
		expect(actual).toStrictEqual('guys');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('index');
		expect(actual).toStrictEqual('indices');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('matrix');
		expect(actual).toStrictEqual('matrices');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('vertex');
		expect(actual).toStrictEqual('vertices');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('canvas');
		expect(actual).toStrictEqual('canvases');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('campus');
		expect(actual).toStrictEqual('campuses');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('campuses');
		expect(actual).toStrictEqual('campuses');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('criterion');
		expect(actual).toStrictEqual('criteria');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('criteria');
		expect(actual).toStrictEqual('criteria');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('genus');
		expect(actual).toStrictEqual('genera');
		expect(actual).toMatchSnapshot();

		actual = inflection.pluralize('genera');
		expect(actual).toStrictEqual('genera');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .singularize', function ()
{
	it('should singularize the given word', function ()
	{
		let actual: string;
		actual = inflection.singularize('status');
		expect(actual).toStrictEqual('status');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('child');
		expect(actual).toStrictEqual('child');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('children');
		expect(actual).toStrictEqual('child');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('address');
		expect(actual).toStrictEqual('address');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('man');
		expect(actual).toStrictEqual('man');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('woman');
		expect(actual).toStrictEqual('woman');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('women');
		expect(actual).toStrictEqual('woman');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('person');
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('people');
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('movies');
		expect(actual).toStrictEqual('movie');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('queries');
		expect(actual).toStrictEqual('query');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('octopi');
		expect(actual).toStrictEqual('octopus');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('Hats');
		expect(actual).toStrictEqual('Hat');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('lives');
		expect(actual).toStrictEqual('life');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('baths');
		expect(actual).toStrictEqual('bath');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('calves');
		expect(actual).toStrictEqual('calf');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('feet');
		expect(actual).toStrictEqual('foot');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('books');
		expect(actual).toStrictEqual('book');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('geese');
		expect(actual).toStrictEqual('goose');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('teeth');
		expect(actual).toStrictEqual('tooth');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('tooth');
		expect(actual).toStrictEqual('tooth');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('knives');
		expect(actual).toStrictEqual('knife');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('halves');
		expect(actual).toStrictEqual('half');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('caves');
		expect(actual).toStrictEqual('cave');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('saves');
		expect(actual).toStrictEqual('save');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('street');
		expect(actual).toStrictEqual('street');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('streets');
		expect(actual).toStrictEqual('street');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('data');
		expect(actual).toStrictEqual('datum');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('meta');
		expect(actual).toStrictEqual('metum');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('whereases');
		expect(actual).toStrictEqual('whereas');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('guys', 'person');
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('matrices');
		expect(actual).toStrictEqual('matrix');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('vertices');
		expect(actual).toStrictEqual('vertex');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('canvases');
		expect(actual).toStrictEqual('canvas');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('campuses');
		expect(actual).toStrictEqual('campus');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('campus');
		expect(actual).toStrictEqual('campus');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('criteria');
		expect(actual).toStrictEqual('criterion');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('criterion');
		expect(actual).toStrictEqual('criterion');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('genera');
		expect(actual).toStrictEqual('genus');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('genus');
		expect(actual).toStrictEqual('genus');
		expect(actual).toMatchSnapshot();

		actual = inflection.singularize('minus');
		expect(actual).toStrictEqual('minus');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .inflect', function ()
{
	it('should correctly inflect the given word based on an integer', function ()
	{
		let actual: string;
		// zero should use plural state
		actual = inflection.inflect('people', 0);
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('men', 0);
		expect(actual).toStrictEqual('men');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('person', 0);
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('octopus', 0);
		expect(actual).toStrictEqual('octopi');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('Hat', 0);
		expect(actual).toStrictEqual('Hats');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('data', 0);
		expect(actual).toStrictEqual('data');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('meta', 0);
		expect(actual).toStrictEqual('meta');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('person', 0, 'guy', 'guys');
		expect(actual).toStrictEqual('guys');
		expect(actual).toMatchSnapshot();

		// greater than 1 should use plural state
		actual = inflection.inflect('people', 2);
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('men', 2);
		expect(actual).toStrictEqual('men');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('person', 2);
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('octopus', 2);
		expect(actual).toStrictEqual('octopi');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('Hat', 2);
		expect(actual).toStrictEqual('Hats');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('data', 2);
		expect(actual).toStrictEqual('data');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('meta', 2);
		expect(actual).toStrictEqual('meta');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('person', 2, 'guy', 'guys');
		expect(actual).toStrictEqual('guys');
		expect(actual).toMatchSnapshot();

		// 1 should use singular state
		actual = inflection.inflect('status', 1);
		expect(actual).toStrictEqual('status');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('child', 1);
		expect(actual).toStrictEqual('child');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('children', 1);
		expect(actual).toStrictEqual('child');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('address', 1);
		expect(actual).toStrictEqual('address');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('person', 1);
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('people', 1);
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('movies', 1);
		expect(actual).toStrictEqual('movie');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('queries', 1);
		expect(actual).toStrictEqual('query');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('octopi', 1);
		expect(actual).toStrictEqual('octopus');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('Hats', 1);
		expect(actual).toStrictEqual('Hat');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('data', 1);
		expect(actual).toStrictEqual('datum');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('meta', 1);
		expect(actual).toStrictEqual('metum');
		expect(actual).toMatchSnapshot();

		actual = inflection.inflect('guys', 1, 'person', 'people');
		expect(actual).toStrictEqual('person');
		expect(actual).toMatchSnapshot();

		// not a number should return original value
		actual = inflection.inflect('original', 'not a number');
		expect(actual).toStrictEqual('original');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .camelize', function ()
{
	it('should camelize the given word', function ()
	{
		let actual: string;
		actual = inflection.camelize('message_properties');
		expect(actual).toStrictEqual('MessageProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('message_properties', true);
		expect(actual).toStrictEqual('messageProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('Message_Properties');
		expect(actual).toStrictEqual('MessageProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('Message_Properties', true);
		expect(actual).toStrictEqual('messageProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('MESSAGE_PROPERTIES');
		expect(actual).toStrictEqual('MESSAGEProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('MESSAGE_PROPERTIES', true);
		expect(actual).toStrictEqual('mESSAGEProperties');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('fooBar_Baz', true);
		expect(actual).toStrictEqual('fooBarBaz');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('FooBar_Baz', true);
		expect(actual).toStrictEqual('fooBarBaz');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('fooBar_fooBaz', true);
		expect(actual).toStrictEqual('fooBarFoobaz');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('FooBar_FooBaz', true);
		expect(actual).toStrictEqual('fooBarFoobaz');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('FooBar');
		expect(actual).toStrictEqual('FooBar');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('FooBar', true);
		expect(actual).toStrictEqual('fooBar');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('Foo/Bar', true);
		expect(actual).toStrictEqual('foo::Bar');
		expect(actual).toMatchSnapshot();

		actual = inflection.camelize('Foo/Bar');
		expect(actual).toStrictEqual('Foo::Bar');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .underscore', function ()
{
	it('should transform the given word with underscore', function ()
	{
		let actual: string;
		actual = inflection.underscore('MessageProperties');
		expect(actual).toStrictEqual('message_properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.underscore('messageProperties');
		expect(actual).toStrictEqual('message_properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.underscore('MP');
		expect(actual).toStrictEqual('m_p');
		expect(actual).toMatchSnapshot();

		actual = inflection.underscore('MP', true);
		expect(actual).toStrictEqual('MP');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .humanize', function ()
{
	it('should humanize the given word', function ()
	{
		let actual: string;
		actual = inflection.humanize('message_properties');
		expect(actual).toStrictEqual('Message properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.humanize('message_properties', true);
		expect(actual).toStrictEqual('message properties');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .capitalize', function ()
{
	it('should capitalize the given word', function ()
	{
		let actual: string;
		actual = inflection.capitalize('message_properties');
		expect(actual).toStrictEqual('Message_properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.capitalize('message properties');
		expect(actual).toStrictEqual('Message properties');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .dasherize', function ()
{
	it('should dasherize the given word', function ()
	{
		let actual: string;
		actual = inflection.dasherize('message_properties');
		expect(actual).toStrictEqual('message-properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.dasherize('Message Properties');
		expect(actual).toStrictEqual('Message-Properties');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .titleize', function ()
{
	it('should titleize the given word', function ()
	{
		let actual: string;
		actual = inflection.titleize('message_properties');
		expect(actual).toStrictEqual('Message Properties');
		expect(actual).toMatchSnapshot();

		actual = inflection.titleize('message properties to keep');
		expect(actual).toStrictEqual('Message Properties to Keep');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .demodulize', function ()
{
	it('should demodulize the given word', function ()
	{
		let actual: string;
		actual = inflection.demodulize('Message::Bus::Properties');
		expect(actual).toStrictEqual('Properties');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .tableize', function ()
{
	it('should tableize the given word', function ()
	{
		let actual: string;
		actual = inflection.tableize('people');
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.tableize('MessageBusProperty');
		expect(actual).toStrictEqual('message_bus_properties');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .kebabize', function ()
{
	it('should kebabize the given word', function ()
	{
		let actual: string;
		actual = inflection.kebabize('people');
		expect(actual).toStrictEqual('people');
		expect(actual).toMatchSnapshot();

		actual = inflection.kebabize('MessageBusProperty');
		expect(actual).toStrictEqual('message-bus-property');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .classify', function ()
{
	it('should classify the given word', function ()
	{
		let actual: string;
		actual = inflection.classify('message_bus_properties');
		expect(actual).toStrictEqual('MessageBusProperty');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .foreign_key', function ()
{
	it('should transform the given word to foreign_key', function ()
	{
		let actual: string;
		actual = inflection.foreign_key('MessageBusProperty');
		expect(actual).toStrictEqual('message_bus_property_id');
		expect(actual).toMatchSnapshot();

		actual = inflection.foreign_key('MessageBusProperty', true);
		expect(actual).toStrictEqual('message_bus_propertyid');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .ordinalize', function ()
{
	it('should ordinalize the given word', function ()
	{
		let actual: string;
		actual = inflection.ordinalize('the 1 pitch');
		expect(actual).toStrictEqual('the 1st pitch');
		expect(actual).toMatchSnapshot();

	});
});

describe('test .transform', function ()
{

	it('should transform the given word with given methods', function ()
	{
		let actual: string;
		actual = inflection.transform('all job', ['pluralize', 'dasherize']);
		expect(actual).toStrictEqual('all-jobs');
		expect(actual).toMatchSnapshot();

		// @ts-ignore
		actual = inflection.transform('all job', ['pluralize', 'anInvalidMethod']);
		expect(actual).toStrictEqual('all jobs');
		expect(actual).toMatchSnapshot();

		actual = inflection.transform('all job', ['capitalize', 'pluralize', 'dasherize']);
		expect(actual).toStrictEqual('All-jobs');
		expect(actual).toMatchSnapshot();

	});

	it('transform allow use function in methods', function ()
	{
		let actual: string;
		actual = inflection.transform('all job', ['capitalize', 'pluralize', 'dasherize', (s) => s + '99']);
		expect(actual).toStrictEqual('All-jobs99');
		expect(actual).toMatchSnapshot();

	});

});
