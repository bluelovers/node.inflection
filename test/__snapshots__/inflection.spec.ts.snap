// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test .camelize should camelize the given word 1`] = `"MessageProperties"`;

exports[`test .camelize should camelize the given word 2`] = `"messageProperties"`;

exports[`test .camelize should camelize the given word 3`] = `"MessageProperties"`;

exports[`test .camelize should camelize the given word 4`] = `"messageProperties"`;

exports[`test .camelize should camelize the given word 5`] = `"MESSAGEProperties"`;

exports[`test .camelize should camelize the given word 6`] = `"mESSAGEProperties"`;

exports[`test .camelize should camelize the given word 7`] = `"fooBarBaz"`;

exports[`test .camelize should camelize the given word 8`] = `"fooBarBaz"`;

exports[`test .camelize should camelize the given word 9`] = `"fooBarFoobaz"`;

exports[`test .camelize should camelize the given word 10`] = `"fooBarFoobaz"`;

exports[`test .camelize should camelize the given word 11`] = `"FooBar"`;

exports[`test .camelize should camelize the given word 12`] = `"fooBar"`;

exports[`test .camelize should camelize the given word 13`] = `"foo::Bar"`;

exports[`test .camelize should camelize the given word 14`] = `"Foo::Bar"`;

exports[`test .capitalize should capitalize the given word 1`] = `"Message_properties"`;

exports[`test .capitalize should capitalize the given word 2`] = `"Message properties"`;

exports[`test .classify should classify the given word 1`] = `"MessageBusProperty"`;

exports[`test .dasherize should dasherize the given word 1`] = `"message-properties"`;

exports[`test .dasherize should dasherize the given word 2`] = `"Message-Properties"`;

exports[`test .demodulize should demodulize the given word 1`] = `"Properties"`;

exports[`test .foreign_key should transform the given word to foreign_key 1`] = `"message_bus_property_id"`;

exports[`test .foreign_key should transform the given word to foreign_key 2`] = `"message_bus_propertyid"`;

exports[`test .humanize should humanize the given word 1`] = `"Message properties"`;

exports[`test .humanize should humanize the given word 2`] = `"message properties"`;

exports[`test .indexOf should return proper index key 1`] = `-1`;

exports[`test .indexOf should return proper index key 2`] = `0`;

exports[`test .inflect should correctly inflect the given word based on an integer 1`] = `"people"`;

exports[`test .inflect should correctly inflect the given word based on an integer 2`] = `"men"`;

exports[`test .inflect should correctly inflect the given word based on an integer 3`] = `"people"`;

exports[`test .inflect should correctly inflect the given word based on an integer 4`] = `"octopi"`;

exports[`test .inflect should correctly inflect the given word based on an integer 5`] = `"Hats"`;

exports[`test .inflect should correctly inflect the given word based on an integer 6`] = `"data"`;

exports[`test .inflect should correctly inflect the given word based on an integer 7`] = `"meta"`;

exports[`test .inflect should correctly inflect the given word based on an integer 8`] = `"guys"`;

exports[`test .inflect should correctly inflect the given word based on an integer 9`] = `"people"`;

exports[`test .inflect should correctly inflect the given word based on an integer 10`] = `"men"`;

exports[`test .inflect should correctly inflect the given word based on an integer 11`] = `"people"`;

exports[`test .inflect should correctly inflect the given word based on an integer 12`] = `"octopi"`;

exports[`test .inflect should correctly inflect the given word based on an integer 13`] = `"Hats"`;

exports[`test .inflect should correctly inflect the given word based on an integer 14`] = `"data"`;

exports[`test .inflect should correctly inflect the given word based on an integer 15`] = `"meta"`;

exports[`test .inflect should correctly inflect the given word based on an integer 16`] = `"guys"`;

exports[`test .inflect should correctly inflect the given word based on an integer 17`] = `"status"`;

exports[`test .inflect should correctly inflect the given word based on an integer 18`] = `"child"`;

exports[`test .inflect should correctly inflect the given word based on an integer 19`] = `"child"`;

exports[`test .inflect should correctly inflect the given word based on an integer 20`] = `"address"`;

exports[`test .inflect should correctly inflect the given word based on an integer 21`] = `"person"`;

exports[`test .inflect should correctly inflect the given word based on an integer 22`] = `"person"`;

exports[`test .inflect should correctly inflect the given word based on an integer 23`] = `"movie"`;

exports[`test .inflect should correctly inflect the given word based on an integer 24`] = `"query"`;

exports[`test .inflect should correctly inflect the given word based on an integer 25`] = `"octopus"`;

exports[`test .inflect should correctly inflect the given word based on an integer 26`] = `"Hat"`;

exports[`test .inflect should correctly inflect the given word based on an integer 27`] = `"datum"`;

exports[`test .inflect should correctly inflect the given word based on an integer 28`] = `"metum"`;

exports[`test .inflect should correctly inflect the given word based on an integer 29`] = `"person"`;

exports[`test .inflect should correctly inflect the given word based on an integer 30`] = `"original"`;

exports[`test .kebabize should kebabize the given word 1`] = `"people"`;

exports[`test .kebabize should kebabize the given word 2`] = `"message-bus-property"`;

exports[`test .ordinalize should ordinalize the given word 1`] = `"the 1st pitch"`;

exports[`test .pluralize should pluralize the given word 1`] = `"people"`;

exports[`test .pluralize should pluralize the given word 2`] = `"men"`;

exports[`test .pluralize should pluralize the given word 3`] = `"women"`;

exports[`test .pluralize should pluralize the given word 4`] = `"women"`;

exports[`test .pluralize should pluralize the given word 5`] = `"people"`;

exports[`test .pluralize should pluralize the given word 6`] = `"octopi"`;

exports[`test .pluralize should pluralize the given word 7`] = `"humans"`;

exports[`test .pluralize should pluralize the given word 8`] = `"aircraft"`;

exports[`test .pluralize should pluralize the given word 9`] = `"luck"`;

exports[`test .pluralize should pluralize the given word 10`] = `"Hats"`;

exports[`test .pluralize should pluralize the given word 11`] = `"lives"`;

exports[`test .pluralize should pluralize the given word 12`] = `"baths"`;

exports[`test .pluralize should pluralize the given word 13`] = `"calves"`;

exports[`test .pluralize should pluralize the given word 14`] = `"feet"`;

exports[`test .pluralize should pluralize the given word 15`] = `"books"`;

exports[`test .pluralize should pluralize the given word 16`] = `"geese"`;

exports[`test .pluralize should pluralize the given word 17`] = `"teeth"`;

exports[`test .pluralize should pluralize the given word 18`] = `"teeth"`;

exports[`test .pluralize should pluralize the given word 19`] = `"knives"`;

exports[`test .pluralize should pluralize the given word 20`] = `"halves"`;

exports[`test .pluralize should pluralize the given word 21`] = `"caves"`;

exports[`test .pluralize should pluralize the given word 22`] = `"saves"`;

exports[`test .pluralize should pluralize the given word 23`] = `"streets"`;

exports[`test .pluralize should pluralize the given word 24`] = `"streets"`;

exports[`test .pluralize should pluralize the given word 25`] = `"data"`;

exports[`test .pluralize should pluralize the given word 26`] = `"meta"`;

exports[`test .pluralize should pluralize the given word 27`] = `"summonses"`;

exports[`test .pluralize should pluralize the given word 28`] = `"whereases"`;

exports[`test .pluralize should pluralize the given word 29`] = `"guys"`;

exports[`test .pluralize should pluralize the given word 30`] = `"indices"`;

exports[`test .pluralize should pluralize the given word 31`] = `"matrices"`;

exports[`test .pluralize should pluralize the given word 32`] = `"vertices"`;

exports[`test .pluralize should pluralize the given word 33`] = `"canvases"`;

exports[`test .pluralize should pluralize the given word 34`] = `"campuses"`;

exports[`test .pluralize should pluralize the given word 35`] = `"campuses"`;

exports[`test .pluralize should pluralize the given word 36`] = `"criteria"`;

exports[`test .pluralize should pluralize the given word 37`] = `"criteria"`;

exports[`test .pluralize should pluralize the given word 38`] = `"genera"`;

exports[`test .pluralize should pluralize the given word 39`] = `"genera"`;

exports[`test .singularize should singularize the given word 1`] = `"status"`;

exports[`test .singularize should singularize the given word 2`] = `"child"`;

exports[`test .singularize should singularize the given word 3`] = `"child"`;

exports[`test .singularize should singularize the given word 4`] = `"address"`;

exports[`test .singularize should singularize the given word 5`] = `"man"`;

exports[`test .singularize should singularize the given word 6`] = `"woman"`;

exports[`test .singularize should singularize the given word 7`] = `"woman"`;

exports[`test .singularize should singularize the given word 8`] = `"person"`;

exports[`test .singularize should singularize the given word 9`] = `"person"`;

exports[`test .singularize should singularize the given word 10`] = `"movie"`;

exports[`test .singularize should singularize the given word 11`] = `"query"`;

exports[`test .singularize should singularize the given word 12`] = `"octopus"`;

exports[`test .singularize should singularize the given word 13`] = `"Hat"`;

exports[`test .singularize should singularize the given word 14`] = `"life"`;

exports[`test .singularize should singularize the given word 15`] = `"bath"`;

exports[`test .singularize should singularize the given word 16`] = `"calf"`;

exports[`test .singularize should singularize the given word 17`] = `"foot"`;

exports[`test .singularize should singularize the given word 18`] = `"book"`;

exports[`test .singularize should singularize the given word 19`] = `"goose"`;

exports[`test .singularize should singularize the given word 20`] = `"tooth"`;

exports[`test .singularize should singularize the given word 21`] = `"tooth"`;

exports[`test .singularize should singularize the given word 22`] = `"knife"`;

exports[`test .singularize should singularize the given word 23`] = `"half"`;

exports[`test .singularize should singularize the given word 24`] = `"cave"`;

exports[`test .singularize should singularize the given word 25`] = `"save"`;

exports[`test .singularize should singularize the given word 26`] = `"street"`;

exports[`test .singularize should singularize the given word 27`] = `"street"`;

exports[`test .singularize should singularize the given word 28`] = `"datum"`;

exports[`test .singularize should singularize the given word 29`] = `"metum"`;

exports[`test .singularize should singularize the given word 30`] = `"whereas"`;

exports[`test .singularize should singularize the given word 31`] = `"person"`;

exports[`test .singularize should singularize the given word 32`] = `"matrix"`;

exports[`test .singularize should singularize the given word 33`] = `"vertex"`;

exports[`test .singularize should singularize the given word 34`] = `"canvas"`;

exports[`test .singularize should singularize the given word 35`] = `"campus"`;

exports[`test .singularize should singularize the given word 36`] = `"campus"`;

exports[`test .singularize should singularize the given word 37`] = `"criterion"`;

exports[`test .singularize should singularize the given word 38`] = `"criterion"`;

exports[`test .singularize should singularize the given word 39`] = `"genus"`;

exports[`test .singularize should singularize the given word 40`] = `"genus"`;

exports[`test .singularize should singularize the given word 41`] = `"minus"`;

exports[`test .tableize should tableize the given word 1`] = `"people"`;

exports[`test .tableize should tableize the given word 2`] = `"message_bus_properties"`;

exports[`test .titleize should titleize the given word 1`] = `"Message Properties"`;

exports[`test .titleize should titleize the given word 2`] = `"Message Properties to Keep"`;

exports[`test .transform should transform the given word with given methods 1`] = `"all-jobs"`;

exports[`test .transform should transform the given word with given methods 2`] = `"all jobs"`;

exports[`test .transform should transform the given word with given methods 3`] = `"All-jobs"`;

exports[`test .transform transform allow use function in methods 1`] = `"All-jobs99"`;

exports[`test .underscore should transform the given word with underscore 1`] = `"message_properties"`;

exports[`test .underscore should transform the given word with underscore 2`] = `"message_properties"`;

exports[`test .underscore should transform the given word with underscore 3`] = `"m_p"`;

exports[`test .underscore should transform the given word with underscore 4`] = `"MP"`;
